src/backend/routers/auth.py

```diff
@@ -133,17 +136,23 @@ async def oidc_authenticate(request: Request, session: DBSessionDep):
 
 
 @router.get("/logout", response_model=Logout)
-async def logout(request: Request):
+async def logout(
+    request: Request,
+    session: DBSessionDep,
+    token: dict | None = Depends(validate_authorization),
+):
     """
-    Logs out the current user.
+    Logs out the current user, adding the given JWT token to the blacklist.
 
     Args:
         request (Request): current Request object.
 
     Returns:
         dict: Empty on success
     """
-    # TODO: Design blacklist
+    if token is not None:
+        db_blacklist = Blacklist(token_id=token["jti"])
```

Comment:
what is the `token` query param defined in this router function suppose to be? i see nothing is passed in there with the unit test https://github.com/cohere-ai/cohere-toolkit/pull/202/files#diff-2ea68412ab67e98ac5c98863b626382b19b598be7396e5db65c96c17d69f882cR93. not sure if im misunderstanding fastapi syntax here 😅

Reply:
The `validate_authorization` dependency returns a decoded token in this case

Reply:
It returns a dict or None, so we still need to validate here `if token is not None:`, then adds it to the blacklist